_modules/variables.md

```diff
@@ -9,4 +9,66 @@ order: 4
+
+```c
+int myFavoriteNumber = 8;
+```
+
+After I have _declared_ `myFavoriteNumber`, I can use it anywhere in my program. And if my favorite number changes later, I can change my variable to reflect that:
+
+```c
+myFavoriteNumber = 21;
+```
+
+Notice that I only have to tell the _compiler_ what type of variable myFavoriteNumber is when I first declare it. If I change the value later, I don't have to include the type; the _compiler_ will remember it automatically. `myFavoriteNumber` is an `int` type variable, which means that it can only be a whole number. If my favorite number was 3.14, this type wouldn't work! Some other common variable types include `double` (for decimal numbers), `bool` (for `true` or `false` values), and `char` (for single characters like the letter `'a'`).
+
+## Examples
+
+Let's add a few variables to our code. In your `variables_activity.cpp` file, in the `variables_exploration` function definition, add the following lines of code:
+
+```c
+int currentLed = 1;
+int currentNote = NOTE_C4;
+int duration = 1000;
+```
+
+We've now made three variables. Hopefully it's obvious what they represent based on their names. Notice that we can set our variables equal to other variables! `currentNote` has the same value as `NOTE_C4`. If we add `speaker_play_note(currentNote, duration)` to our code, it's the same as adding `speaker_play_note(NOTE_C4, 1000)`. Middle C will be played for one second.
+
+Let's use our variables to do something interesting. First, let's light up our current LED. Add the following to your code, and upload it to your board:
+
+```c
+leds_set_color(currentLed, 255, 0, 0);
+```
+
+It's nice that we don't have to hard-code the number 1 into our code to tell the _compiler_ that we want to light it up. It makes our code a little easier to read. We can even use it to light up other leds. Add the following to your code and reupload it to your board:
+
+```c
+currentLed++;
+leds_set_color(currentLed, 255, 0, 0);
+currentLed++;
+leds_set_color(currentLed, 255, 0, 0);
+currentLed++;
+leds_set_color(currentLed, 255, 0, 0);
+```
+
+As you probably guessed by the lights on your board turning on, adding `++` to our variable name (with a semicolon at the end) increments the number our variable represents by 1. We've now turned on the first four LEDs on without hard-coding their numbers anywhere!
+
+## Exploration
+
+1. Divide `currentNote` by 2 using the following code snippet: `currentNote = currentNote / 2;` Play a sound again. How does our new sound compare to our old one?
+
+1. See what happens if you multiply or divide `currentNote` by powers of 2 and use it to play a sound. For example, how does the sound compare to our old ones if we write `currentNote = currentNote * 8;`?
```

Comment:
... by powers of 2 (2, 4, 8, 16) and use it ...

Reply:
Done